src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-management.component.ts

```diff
@@ -34,8 +34,11 @@ export class LectureUnitManagementComponent implements OnInit, OnDestroy {
     lecture: Lecture;
     isLoading = false;
     updateOrderSubject: Subject<any>;
+    viewButtonAvailable = new Map<number, boolean>();
```

Comment:
Calling `get` on a map in the template is still a method call. Maybe use a record instead of a map:

```suggestion
    viewButtonAvailable: Record<number, boolean> = {};
```

You'd set a value using

```ts
this.viewButtonAvailable[lectureUnit.id!] = this.isViewButtonAvailable(lectureUnit);
```

and in the template, you can use

```ts
@if (viewButtonAvailable[lectureUnit.id!]) {

}
```

to check the value

Reply:
Changed